lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1744,6 +1751,9 @@ class SemanticsObject {
       hasFlag(ui.SemanticsFlag.hasCheckedState) ||
       hasFlag(ui.SemanticsFlag.hasToggledState);
 
+  bool get isSelectable =>
+      hasFlag(ui.SemanticsFlag.isSelected);
```

Comment:
This probably won't work. "selectED" is different from "selectABLE". When `isSelected` is false, the node may still be selectable, it's just not selected right now, but the current logic in this PR will not make the node selectable, e.g. it will not set `aria-selected="false"`. Another problem is that when `isSelected` changes from `false` to `true`, the node will change its role, which will confuse the browser and the screen reader, possibly losing element focus.

I see two clean ways to make this work:

1. We introduce `SemanticsFlag.hasSelectedState`, similar to `hasToggledState`, `hasCheckedState`, and `hasEnabledState`. If `hasSelectedState` is true, then we add the `SemanticSelectable` behavior.
2. We enumerate all roles (adding missing roles, such as `tab`, if necessary) that we know are selectable. Then we add the `SemanticSelectable` behavior to those roles.